sdk/core/core-client-lro-rest/review/core-client-lro.api.md

```diff
@@ -0,0 +1,26 @@
+## API Report File for "@azure-rest/core-client-lro"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { Client } from '@azure-rest/core-client';
+import { HttpResponse } from '@azure-rest/core-client';
+import { LroEngineOptions } from '@azure/core-lro';
+import { PollerLike } from '@azure/core-lro';
+import { PollOperationState } from '@azure/core-lro';
+
+// @public
+export function getLongRunningPoller<TResult extends HttpResponse>(client: Client, initialResponse: TResult): PollerLike<PollOperationState<TResult>, TResult>;
```

Comment:
`getLongRunningPoller` feels a bit verbose, did you consider using just `getPoller`? cc @bterlson @xirzec @chradek

Reply:
Initially, I named it `getPoller` but when prototyping how this would be generated I decided to name it like this to avoid collisions, for example, this is how the generator would consume this to export `getPoller`

```typescript
export function getPoller<TResult extends HttpResponse>(
  client: FarmBeatsRestClient,
  initialResponse: TResult
): PollerLike<PollOperationState<TResult>, TResult> {
  return getLongRunningPoller(client, initialResponse);
}
```

If this one is named `getPoller` as well we'd need an import alias. Not a big deal though, I'm happy to change it to `getPoller` I'll wait to get more opinions from @bterlson and @xirzec

Reply:
But why generating this wrapper? just to hide the pollerOptions? but again, why?

Reply:
Discussed offline. This wrapper is generated to pass additional info to the LRO engine that we can abstract from the user. For example, there are cases where the Swagger would define an LRO that needs FinalStateVia, we would set that in this wrapper. The end-user doesn't need to know about FinalStateVia, they would only care about the polling interval and also a state to resume the polling if they want. The other options are meant for internal use only. So this wrapper hides that from end users.